user/APP/CAN_receive/CAN_receive.c

```diff
@@ -35,17 +35,26 @@
 
 /******************** Private Defines for Receiving ********************/
 
-#define get_motor_measure(ptr, rx_message)                                                     \
+#define fill_motor_readings(ptr, rx_message)                                                     \
     {                                                                                          \
         (ptr)->last_ecd = (ptr)->ecd;                                                          \
         (ptr)->ecd = (uint16_t)((rx_message)->Data[0] << 8 | (rx_message)->Data[1]);           \
         (ptr)->speed_rpm = (uint16_t)((rx_message)->Data[2] << 8 | (rx_message)->Data[3]);     \
-        (ptr)->given_current = (uint16_t)((rx_message)->Data[4] << 8 | (rx_message)->Data[5]); \
+        (ptr)->current_read = (uint16_t)((rx_message)->Data[4] << 8 | (rx_message)->Data[5]); \
         (ptr)->temperate = (rx_message)->Data[6];                                              \
     }
 
-
-		/// TODO: Make similar get_motor_measure for M3508 and P36
+#define fill_gimbal_motor_readings(ptr, rx_message)                                              \
+    {                                                                                          \
+        (ptr)->last_ecd = (ptr)->ecd;                                                          \
+        (ptr)->ecd = (uint16_t)((rx_message)->Data[0] << 8 | (rx_message)->Data[1]);           \
+        (ptr)->current_read = (uint16_t)((rx_message)->Data[2] << 8 | (rx_message)->Data[3]); \
+        (ptr)->speed_rpm = (uint16_t)((rx_message)->Data[4] << 8 | (rx_message)->Data[5]);     \
+        (ptr)->temperate = (rx_message)->Data[6];                                              \
+    }
```

Comment:
Which parts of the message are speed_rpm and current_read? - If someone has a motor this can easily be tested (again).